test/support/mock_fs_helper.ex

```diff
@@ -0,0 +1,49 @@
+defmodule MixTestWatch.MockFsHelper do
```

Comment:
I am a bit confused. I assumed you wanted real files, created on a case-by-case basis. Or do you want a file structure with some empty files being checked into git, spanning all cases?

The latter alarms be a bit because it blurs the separation between tests.